realm-jni/src/io_realm_internal_table.cpp

```diff
@@ -1389,9 +1389,10 @@ JNIEXPORT jlong JNICALL Java_io_realm_internal_Table_createNative(JNIEnv *env, j
 
 // Checks if the primary key column contains any duplicate values, making it ineligible as a
 // primary key.
-bool check_valid_primary_key_column(JNIEnv* env, Table* table, size_t column_index) // throws
+bool check_valid_primary_key_column(JNIEnv* env, Table* table, StringData column_name) // throws
 {
-    int column_type = table->get_column_type(column_index);
+    size_t column_index = table->get_column_index(column_name);
+    DataType column_type = table->get_column_type(column_index);
```

Comment:
You have to check the index returned is valid before using it, and throwing if it is invalid.